docs/semgrep-ci/configuring-blocking-and-errors-in-ci.md

```diff
@@ -3,59 +3,63 @@ slug: configuring-blocking-and-errors-in-ci
+When Semgrep identifies one or more blocking findings, it returns exit code `1`. You can use this result to set up additional checks to enforce a block in your CI/CD pipeline, such as not allowing the merge of the PR/MR. This action applies to both full scans and [diff-aware scans](/semgrep-code/glossary#diff-aware-scan).
 
-:::tip
 The process to enforce a block on a PR or MR after Semgrep exits with error code `1` is dependent on your CI provider. Review your CI provider's documentation for further information.
+
+If Semgrep encounters an internal error, it sends an anonymous crash report to a crash-reporting server and returns exit code `0` to prevent blocking your CI pipeline.
+
+:::info
+See [CLI reference](/cli-reference#exit-codes) for more information about Semgrep exit codes.
 :::
 
 ## Configuration options for blocking findings and errors
+ 
+You can configure, change, or revert to the default setup of blocking findings and errors in your CI pipeline by passing one of the following options in the `semgrep.yml` file used to configure and run Semgrep in your CI pipeline:
 
-Configure, change or revert to the default setup of blocking findings and errors in your CI pipeline using the following options in CI configuration file:
-
-| CI option                                      | Description                         |
+| CI option | Description |
 |------------------------------------------------|-------------------------------------|
-| `semgrep ci` or `semgrep ci --suppress-errors` | Default: CI **fails** on blocking findings, CI **passes** on internal errors.  |
-| `semgrep ci --no-suppress-errors`              | CI **fails** on blocking findings, CI **fails** on internal errors.            |
-| <code>semgrep ci &vert;&vert; true</code>      | CI **passes** on blocking findings, CI **passes** on internal errors.          |
+| `semgrep ci` or `semgrep ci --suppress-errors` | Default. CI **fails** on blocking findings, but **passes** on internal errors. |
+| `semgrep ci --no-suppress-errors` | CI **fails** on blocking findings and internal errors. |
+| <code>semgrep ci &vert;&vert; true</code> | CI **passes** on blocking findings and internal errors. |
 
-To change this configuration, insert one of the configuration options (flags) after the following keys in CI YAML configuration file of Semgrep:
-- On GitHub, insert the flag after the `run` key (for example, `run: semgrep ci --suppress-errors` to state the default option).
-- On GitLab, insert the flag after the `script` key (for example, `script: semgrep ci --suppress-errors` to state the default option).
-- Insert these flags in an equivalent key in configuration files of other CI providers.
+To change Semgrep's behavior, modify the `semgrep.yml`, specifically the `semgrep ci` command to the CI option that best fits your needs. For example, GitHub users would include the following under the `run` key:
```

Comment:
```suggestion
To change Semgrep's behavior, modify your pipeline or job file, specifically the `semgrep ci` command, to the CI option that best fits your needs. For example, GitHub users should edit the `semgrep.yml` workflow file and include the following under the `run` key:
```
Not all workflow files are `semgrep.yml`, so I generalized the tidbit and specificed `semgrep.yml` for our github example.